### PR TITLE
Defibs now uses TG's system (mostly). Harmfull defibbing and no weirdness anymore

### PR DESCRIPTION
**Pull request by farie82**
**What does this PR do:**
Uses most of TG's code now. With a little para thrown in.
Combat (or emagged) defibs can now be used to cause heart attacks. Use the defib in harm mode to do this
Borgs use the same path as normal defib paddles

fixes: #11178

**Changelog:**
:cl:
fix: Borg defibs work like normal defibs now.
add: Combat (or emagged) defibs can now be used to cause heart attacks. Use the defib in harm mode to do this
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
The defib refactor in the original newcrit PR is better than this.

utilizing that would be the proper way to do this; this is super hacky and won't be accepted.

**Comment by farie82** [comment]
> The defib refactor in the original newcrit PR is better than this.
> 
> utilizing that would be the proper way to do this; this is super hacky and won't be accepted.

I'll check the method used there more tomorrow maybe. I saw that the parent fixes things for their child though. I'll try to make it more OO

**Comment by farie82** [comment]
@Fox-McCloud changed it to your old code.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -831,8 +831,8 @@ var/list/robot_verbs_default = list(
 				src.module.modules += new /obj/item/pickaxe/drill/cyborg/diamond(src.module)
 				src.module.rebuild()
 			if(src.module && istype(src.module, /obj/item/robot_module/medical))
-				for(var/obj/item/borg_defib/F in src.module.modules)
-					F.safety = 0
+				for(var/obj/item/twohanded/shockpaddles/cyborg/F in src.module.modules)
+					F.combat = 0
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/robot.dm]: this isn't needed anymore.

**Comment by Fox-McCloud** [comment]
I apologize; I should have clarified.

This means defibs won't revive anymore; that's something players probably won't like.

The *basis* of the rework is what you want; it should still revive.

It's based off of TG's refactor of defibs.

now that you have *most* of it though, it should be easy to finish out.

Sorry again, should have been more clear.

**Comment by farie82** [comment]
> I apologize; I should have clarified.
> 
> This means defibs won't revive anymore; that's something players probably won't like.
> 
> The _basis_ of the rework is what you want; it should still revive.
> 
> It's based off of TG's refactor of defibs.
> 
> now that you have _most_ of it though, it should be easy to finish out.
> 
> Sorry again, should have been more clear.

Ah yeah didn't check that. Updated it now so it will actually revive a person.

**Comment by Fox-McCloud** [comment]
Since you're already halfway there, I'd just go ahead and finish this out to TG's defib--that's where I was going with it had newcrit not been implemented.

Back defibs should probably have a delay, given that they revive and that's...considerably more powerful than what the handheld ones do.

**Comment by farie82** [comment]
@Fox-McCloud changed the code so it has most of TG's code in it. It still doesn't have it fully. This is because I don't see the need myself to fully port the rest, mostly because they use another medical system for those parts.

**Comment by Fox-McCloud** [comment]
cyborg defibs were well and truly shitcode.

**Comment by farie82** [comment]
If somebody else wants to take on the PR feel free.

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 2]
```diff
@@ -1,600 +1,600 @@
-/obj/item/robot_module
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: Putting my foot down on these PRs that overwrite the entire file; it's making git blame and history very difficult to navigate.

You'll have to resolve this or resubmit in a version that doesn't have this; I figured github would handled this more gracefully, on merge...but it didn't; files that get overwritten like this show up in the git blame+history; not good at all.

**Comment by farie82** [comment]
I hate git sometimes. Remaking this one in a bit after some testing

**Pull request closed without merging**